Tools/CISupport/ews-build/steps.py

```diff
@@ -213,17 +214,32 @@ def get_reviewers(self, pr_number, repository_url=None):
         if not api_url:
             return []
 
-        reviews_url = f'{api_url}/pulls/{pr_number}/reviews'
-        content = self.fetch_data_from_url_with_authentication_github(reviews_url)
-        if not content:
-            return []
+        reviews = []
+        reviews_url = f'{api_url}/pulls/{pr_number}/reviews?per_page={self.LIMIT}'
+        page = 1
+        while page < 10:
+            content = self.fetch_data_from_url_with_authentication_github(f'{reviews_url}&page={page}')
+            if not content:
+                break
+            response_content = content.json() or []
+            if not isinstance(response_content, list):
+                break
+            reviews += response_content
+            if len(response_content) >= self.LIMIT:
+                continue
+            page += 1
 
-        result = []
-        for review in (content.json() or []):
+        approved = dict()
+        rejected = dict()
+        for review in reviews:
             reviewer = review.get('user', {}).get('login')
-            if reviewer and review.get('state') == 'APPROVED':
-                result.append(reviewer)
-        return result
+            if not reviewer:
+                continue
+            if review.get('state') == 'APPROVED':
+                approved[reviewer][review.get('id', 0)]
+            elif review.get('state') == 'CHANGES_REQUESTED':
+                rejected[reviewer][review.get('id', 0)]
+        return [reviewer for reviewer, id_val in approved.items() if id_val > rejected.get(reviewer, 0)]
```

Comment:
what's the logic being implemented here? what's `id_val`? maybe rename id_val to something more readable.

Reply:
We're trying to make sure that someone can override an r+ with an r- in the future. `id_val` comes from GitHub, it's the review "id" (according to GitHub's API). Larger numbers are newer. We could use timestamp, expect I didn't really want to decode the GitHub timestamp.

Reply:
ok.
Also, id_val is a local variable in this line, maybe we can call it `_id`.
Also, since this is important logic, probably having a explicit for loop with a comment explaining what we are doing is better (instead of using list comprehension).